Sources/Testing/Test.swift

```diff
@@ -210,17 +208,48 @@ extension Test {
     /// - ``Test/parameters``
     public var parameters: [Parameter]?
 
+    /// Serializable representation of the comments associated with this test.
+    // We can't store arbitrary traits on snapshots because they're not
+    // generally encodable, but we can store comment traits.
+    public var comments: [Comment]
+
+    /// Serializable representation of the tags associated with this test.
+    // We can't store arbitrary traits on snapshots because they're not
+    // generally encodable, but we can store tags.
+    public var tags: Set<Tag>
+
+    /// The set of bugs associated with this test.
+    ///
+    /// For information on how to associate a bug with a test, see the
+    /// documentation for ``Bug``.
+    public var associatedBugs: [Bug]
+
+    // Backing storage for ``Test.Snapshot.timeLimit``.
+    private var _timeLimit: TimeValue?
+
+    /// Serializable representation of the ``TimeLimitTrait``
+    @available(_clockAPI, *)
+    public var timeLimit: Duration? {
+      _timeLimit.map(Duration.init)
+    }
+
     /// Initialize an instance of this type by snapshotting the specified test.
     ///
     /// - Parameters:
     ///   - test: The original test to snapshot.
-    public init(snapshotting test: Test) {
+    public init(snapshotting test: borrowing Test) {
       id = test.id
       name = test.name
       displayName = test.displayName
       sourceLocation = test.sourceLocation
       testCases = test.testCases?.map(Test.Case.Snapshot.init)
       parameters = test.parameters
+      comments = test.comments
+      tags = test.tags
+      associatedBugs = test.traits.compactMap { $0 as? Bug }
+      if #available(_clockAPI, *) {
+        _timeLimit = test.timeLimit.map(TimeValue.init)
```

Comment:
Because `_timeLimit` is the stored property, we likely need a custom implementation of `CodingKeys` on this type in order to give it the right key in JSON (`"timeLimit"`, not `"_timeLimit"`.)